pkg/util/test/util.go

```diff
@@ -2002,6 +2002,49 @@ func ValidateTelemetryInstalled(pxImageList map[string]string, cluster *corev1.S
 	return nil
 }
 
+// ValidatePodTopologySpreadConstraints validates pod topology spread constraints for following component deployments:
+// csi, pvc controller, stork, stork controller
+func ValidatePodTopologySpreadConstraints(t *testing.T, namespace string, topologyKeys []string, timeout, interval time.Duration) {
+	topologyKeySet := make(map[string]bool)
+	for _, key := range topologyKeys {
+		topologyKeySet[key] = true
+	}
+
+	validateDeploymentTopology := func(deploymentName string) func() (interface{}, bool, error) {
+		return func() (interface{}, bool, error) {
+			deployment, err := appops.Instance().GetDeployment(deploymentName, namespace)
+			if err != nil {
+				return nil, true, fmt.Errorf("failed to get deployment %s/%s", namespace, deploymentName)
+			}
+			constraintKeySet := make(map[string]bool)
+			for _, constraint := range deployment.Spec.Template.Spec.TopologySpreadConstraints {
+				constraintKeySet[constraint.TopologyKey] = true
+			}
+			if !reflect.DeepEqual(topologyKeySet, constraintKeySet) {
+				return nil, true, fmt.Errorf("failed to validate deployment pod topology spread constraints, expected keys: %s, actual keys: %v", topologyKeys, constraintKeySet)
+			}
+			return nil, false, nil
```

Comment:
Can we additionally validate that the pods are actually spread across the nodes with different fault domains?